06_miscellaneous/floss_licenses_slides.md

```diff
@@ -0,0 +1,173 @@
+---
+type: slide
+slideOptions:
+  transition: slide
+  width: 1400
+  height: 900
+  margin: 0.1
+---
+
+<style>
+  .reveal strong {
+    font-weight: bold;
+    color: orange;
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+  .reveal code {
+    font-family: 'Ubuntu Mono';
+    color: orange;
+  }
+  .reveal section img {
+    background:none;
+    border:none;
+    box-shadow:none;
+  }
+</style>
+
+# FLOSS Licenses
+
+---
+
+## Disclaimer
+
+- IANAL: _I am not a lawyer._
+- Provided information "as-is", no warranty
+- Things are simplified sometimes.
+
+---
+
+## Licenses are Complicated, Right?
+
+- The details, yes
+- The big picture, no
+- Every RSE should know the big picture.
+
+---
+
+## Use Cases
+
+- You want to pick a license for your (new) code.
+- You use software of others and want to know what you are allowed to do.
+- You contribute to an existing software and want to understand what this legally means.
+
+---
+
+## Code Without a License
+
+- Code is creative work and, thus, under **exclusive copyright by default**.
+- Copyright owner is typically the creator (or their employer).
+- Nobody else can copy / distribute / modify
+- If code has more than one contributor, all contributors might become this _"nobody"_.
+- **GitHub terms of service**:
+    - Others can view and fork your repository.
+    - Does not imply that they can use / modify / share your code.
+- Still, you do not have to give a license.
+    - Even though not necessary, putting a copyright notice in a prominent place helps clarifying in this case.
+
+---
+
+## Using a License
+
+- A FLOSS license allows reuse of your code while retaining copyright.
+- To choose or changing a license all copyright owners must agree (if not stated otherwise).
+
+---
+
+## Types of Licenses
+
+... but there are sooo many licenses, they even have different versions. :scream:
+
+Focus on big picture: There are mainly **three types**:
+
+- Permissive licenses (not restrictive)
+- Copyleft licenses (very restrictive)
+- Intermediate licenses (complicated, will not cover here)
+
+... and always choose the newest version of a license.
```

Comment:
Why should one use the newest version? Wasn't there the big issue with GPLv2 vs. GPLv3 at some point? The Linux kernel is also still under GPLv2 (with some extension)

Reply:
Because you cannot update that easily. I guess the Linux kernel is trapped in v2. GNU recommends to choose "GPL v3 or newer" as e.g. [DuMuX does](https://git.iws.uni-stuttgart.de/dumux-repositories/dumux/-/blob/master/LICENSE.md#license) 
A good, but complicated article on this: https://www.gnu.org/licenses/license-compatibility.html

Reply:
Linux sticked to GPLv2 on purpose, see <https://unicef.github.io/inventory/legal-policy/gpl-comparison/>

I do not remember the full argument anymore though. But good to know. Then I know what to say. ;)